csvy/validators/table_schema.py

```diff
@@ -0,0 +1,454 @@
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'datetime'.
+        format_: A string specifying a format. It can be 'default' or 'any', or a string
+            with a custom format parsable by Python / C standard 'strptime', eg.
+            '%d/%m/%y'.
+
+    """
+
+    type_: Literal[TypeEnum.DATETIME, TypeEnum.DATE, TypeEnum.TIME] = Field(
+        TypeEnum.DATETIME, alias="type", description="A string specifying the type."
+    )
+    format_: FormatDateTime | str | None = Field(
+        None,
+        alias="format",
+        description="A string specifying a format. It can be "
+        + "'default' or 'any', or a string with a custom format parsable by Python / C "
+        + "standard 'strptime', eg. '%d/%m/%y'.",
+    )
+
+
+class GeopointColumnValidator(CommonValidator):
+    """Validator for the geopoint columns in the Table Schema.
+
+    This class is used to validate the geopoint columns in the Table Schema. It is based
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'geopoint'.
+        format_: A string specifying a format, with valid values being 'default',
+            'array', 'object' and None.
+
+    """
+
+    type_: Literal[TypeEnum.GEOPOINT] = Field(
+        TypeEnum.GEOPOINT, alias="type", description="A string specifying the type."
+    )
+    format_: FormatGeoPoint | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+
+
+class GeoJSONColumnValidator(CommonValidator):
+    """Validator for the geojson columns in the Table Schema.
+
+    This class is used to validate the geojson columns in the Table Schema. It is based
+    on the columns defined in the Table Schema specification.
+
+    Attributes:
+        type_: A string specifying the type, with valid values being 'geojson'.
+        format_: A string specifying a format, with valid values being 'default',
+            'topojson' and None.
+
+    """
+
+    type_: Literal[TypeEnum.GEOJSON] = Field(
+        TypeEnum.GEOJSON, alias="type", description="A string specifying the type."
+    )
+    format_: FormatGeoJSON | None = Field(
+        None, alias="format", description="A string specifying a format."
+    )
+
+
+ColumnValidator = Annotated[
+    StringColumnValidator
+    | NumberColumnValidator
+    | IntegerColumnValidator
+    | BooleanColumnValidator
+    | DateTimeColumnValidator
+    | GeopointColumnValidator
+    | GeoJSONColumnValidator,
+    Field(discriminator="type_"),
+]
+"""Annotated type for the ColumnValidator."""
+
+
+@register_validator("schema")
+class SchemaValidator(BaseModel):
+    """Validator for the Table Schema in the CSVY file.
+
+    This class is used to validate the Table Schema in the CSVY file. It is based on the
+    schema defined in the Table Schema specification.
```

Comment:
I think we should link to the Table Schema website somewhere, and it seems like this might be the best place for it.

```suggestion
    schema defined in the [Table Schema specification](https://specs.frictionlessdata.io/table-schema/#language).
```

Reply:
Do markdown-style links like this work in mkdocs @AdrianDAlessandro? If so, I'm in favour :smile:

Reply:
I think they should. Having said that, we do not have documentation yet (see https://github.com/ImperialCollegeLondon/pycsvy/issues/170, in case you have free time 😆 ), so we cannot test that. 

In any case, I've added it, because it is true it should be mentioned somewhere.

Reply:
> Do markdown-style links like this work in mkdocs @AdrianDAlessandro? If so, I'm in favour 😄

I'm 99% sure they do. But I think including the link this way for now is the best and then can be checked later once docs are built

Reply:
Done!